convert_to_hdf5/src/convert_to_hdf5.py

```diff
@@ -188,6 +190,56 @@ def load_hdf_files(record_names, out_dir, data_info_mapping, meta_data, topic_in
                 rospy.loginfo('\t\tNo meta info')
     return hdf5_files
 
+def match_depth(hdf5_files, data_info_mapping):
+    upper_color_match_topic = 'vid/color/data/upper/matched_depth_index'
+    lower_color_match_topic = 'vid/color/data/lower/matched_depth_index'
+
+    timestamps = {}
+    dataset_upper = None
+    dataset_lower = None
+
+    for hdf5_database in hdf5_files:
+        if upper_color_match_topic not in hdf5_database :
+            dataset_upper = hdf5_database.create_dataset(upper_color_match_topic, (INITIAL_SIZE,),
+                                         maxshape=(None,),
+                                         dtype=np.uint32, chunks=(CHUNK_SIZE,))
+        else:
+            dataset_upper = hdf5_database[upper_color_match_topic]
+
+        if lower_color_match_topic not in hdf5_database:
+            dataset_lower = hdf5_database.create_dataset(lower_color_match_topic, (INITIAL_SIZE,),
+                                         maxshape=(None,),
+                                         dtype=np.uint32, chunks=(CHUNK_SIZE,))
+        else:
+            dataset_lower = hdf5_database[lower_color_match_topic]
+
+        for vid_topic in data_info_mapping:
+            timestamp_secs = hdf5_database[vid_topic + '/secs']
```

Comment:
It seems like this will through an index error if the topic doesn't have an associated secs and nsecs. Maybe it is best to not read from the database for every topic. Only ones that are known to be videos where we have stored this stuff? We will have other topic types in there. I suspect that they will have secs and nanoseconds too, but best to not read till you know what you are reading.

Reply:
Added a:  if 'vid' in vid_topic here. Should make it extendable for later on